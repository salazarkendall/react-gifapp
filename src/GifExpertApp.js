import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';

// Esto es un Functional Component
const GifExpertApp = () => {
	/**
	 * Normalmente las constantes que necesitamos que se encuentren en otros componentes, los ponemos dentro de hooks
	 * Para poder hacer un hook mucho mas rapido podemos usar el snippet _ush_ el cual nos crea el hook
	 * Nota: para poder usar el hook, debemos importar el useState de React
	 */
	const [categories, setCategories] = useState([
		'Gumball',
		'Hora de aventura',
		'Un show mas',
	]);

	/**
	 * Todos los functional components deben retornar una estructura HTML que sera agregada al DOM
	 * Dentro de estos componentes podemos agregar otros componentes como se muestra a continuacion
	 */
	return (
		<>
			<h1>GifExpertApp</h1>
			<hr />
			<AddCategory setCategories={setCategories} />
			<ol>
				{categories.map((category) => {
					return <li key={category}>{category}</li>;
				})}
			</ol>
		</>
	);
};

export default GifExpertApp;
