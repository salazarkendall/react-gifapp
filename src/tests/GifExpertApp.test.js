/**
 * Las importaciones por defecto no llevan {}
 * Las importaciones comunes deben de llevar {} por defecto
 */

import { shallow } from 'enzyme';
import React from 'react';
import GifExpertApp from '../GifExpertApp';

describe('Pruebas en GifExpertApp', () => {
	let wrapper = shallow(<GifExpertApp />);

	test('Debe de hacer match el shallow con el snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
