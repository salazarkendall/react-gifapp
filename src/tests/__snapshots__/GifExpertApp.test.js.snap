// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp Debe de hacer match el shallow con el snapshot 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <hr />
  <button
    onClick={[Function]}
  >
    Agregar
  </button>
  <ol>
    <li
      key="Gumball"
    >
      Gumball
    </li>
    <li
      key="Hora de aventura"
    >
      Hora de aventura
    </li>
    <li
      key="Un show mas"
    >
      Un show mas
    </li>
  </ol>
</Fragment>
`;
